XenAdmin/Wizards/PatchingWizard/AutomatedUpdatesBasePage.cs

```diff
@@ -140,6 +141,19 @@ protected virtual bool SkipInitialPlanActions(Host host)
         protected virtual void DoAfterInitialPlanActions(UpdateProgressBackgroundWorker bgw, Host host, List<Host> hosts) { }
         #endregion
 
+        private void RearrangeDelayedActionsAfterUpdating(Host host, ref List<PlanAction> delayedActions)
+        {
+            delayedActions = delayedActions.Distinct().ToList();
+
+            if (delayedActions.Any(a => a is RestartHostPlanAction))
+            {
+                if (!Helpers.ElyOrGreater(host) || host.updates_requiring_reboot.Count > 0)

```

Comment:
As discussed, if host.updates_requiring_reboot contains only the failed livepatches and not all patches requiring reboot, there will be no reboot - can we confirm what it contains?

Reply:
needs confirmation from toolstack team.

Reply:
host.updates_requiring_reboot is only for failed live patch application, we need to consider non-live-patch with restartHost separately.